server/controllers/translation.js

```diff
@@ -0,0 +1,73 @@
+const asyncHandler = require("express-async-handler");
+const Conversation = require("../models/Conversation");
+const User = require("../models/User");
+const mongoose = require("mongoose");
+const { Translate } = require("@google-cloud/translate").v2;
+const trans = new Translate({
+  projectId: process.env.GOOGLE_TRANSLATE_PROJECT_ID,
+  KeyFilename: process.env.GOOGLE_APPLICATION_CREDENTIALS,
+});
+
+// @route POST /users/translation/:conversationId
+// translate message from sender's language to
+// all other recipients' languages and save the record
+exports.postMessageTranslation = asyncHandler(async (req, res) => {
+  const conversation_id = req.params.conversationId;
+  const user_id = req.user.id;
+  const text_to_translate = req.body.message;
+  var from_language = "";
+  var to_languages = [];
+
+  const conversation = await Conversation.findById(conversation_id);
+
+  // get sender's primary language
+  // and all other recipients' primary languages
+  for (let uid of conversation.users) {
+    let user = await User.findById(uid);
+    if (uid == user_id) {
```

Comment:
```suggestion
    if (uid === user_id) {
```
You may need to convert types to make this work but that's better practice